app/models/folio/patron.rb

```diff
@@ -125,76 +132,68 @@ def remaining_checkouts
       borrow_limit - checkouts.length
     end
 
-    def proxy_borrower?
-      false
-    end
-
-    def sponsor?
-      false
-    end
-
-    def checkouts
-      @checkouts ||= patron_info['loans'].map { |checkout| Checkout.new(checkout) }
+    def all_fines
+      # @all_fines ||= []
+      @all_fines ||= patron_info['accounts'].map { |fine| Fine.new(fine) }
     end
 
     def fines
       all_fines
       # all_fines.reject { |fine| payment_sequence.include?(fine.sequence) }
     end
 
-    def all_fines
-      # @all_fines ||= []
-      @all_fines ||= patron_info['accounts'].map { |fine| Fine.new(fine) }
+    # TODO: figure this out for FOLIO
+    def group_fines
+      # return fines.select { |fine| group_billuser_info_keys.include?(fine.key) } if sponsor?
+      # fines
+      []
     end
 
-    ##
     # Creates a range of integers based of a payment sequence string
     def payment_sequence
       return Range.new(0, 0) unless payment_in_process[:billseq] && payment_in_process[:pending]
 
       Range.new(*payment_in_process[:billseq].split('-').map(&:to_i))
     end
 
-    def requests
-      @requests ||= folio_requests + borrow_direct_requests
+    def proxy_borrower?
+      user_info['proxiesFor']&.any?
+    end
+
+    def sponsor?
+      user_info['proxiesOf']&.any?
     end
 
     def group
-      @group ||= Folio::Group.new(user_info)
+      @group ||= Folio::Group.new(patron_info)
     end
 
     def group?
       group&.member_list&.any?
     end
 
-    def group_checkouts
-      return checkouts.select { |checkout| group_circuser_info_keys.include?(checkout.key) } if sponsor?
-
-      checkouts
-    end
-
-    def group_circuser_info_keys
-      @group_circuser_info_keys ||= SymphonyDbClient.new.group_circuser_info_keys(key)
+    def all_checkouts
+      @all_checkouts ||= patron_info['loans'].map { |checkout| Checkout.new(checkout) }
     end
 
-    def group_requests
-      return requests.select { |request| group_holduser_info_keys.include?(request.key) } if sponsor?
-
-      requests
+    # Self checkouts
+    def checkouts
+      all_checkouts.reject(&:proxy_checkout?)
     end
 
-    def group_holduser_info_keys
-      @group_holduser_info_keys ||= SymphonyDbClient.new.group_holduser_info_keys(key)
+    # Checkouts from the proxy group
+    def group_checkouts
+      return all_checkouts.select(&:proxy_checkout?) if sponsor?
     end
```

Comment:
```suggestion
    def group_checkouts
      all_checkouts.select(&:proxy_checkout?)
    end
```

Does the return value for non-sponsors matter? Or, if it does, could we return an empty array to simplify any downstream consumers?

Reply:
group_checkouts is only ever called in the `Group` class itself by `Group#checkouts`

```
   def checkouts
      sponsor.group_checkouts
    end
```

So I think the `if sponsor?` return is good. I will add that back.

Reply:
I can't help but foresee a situation where a proxy user is also a sponsor and everything breaks. If it's fine to just return `all_checkouts.select(&:proxy_checkout?)` that seems best, but if there's some reason that'd be bad, at least returning `[]` would keep things from throwing exceptions

Reply:
I think it would be fine. This is consumed in the `checkouts_controller`:
```
    @checkouts = patron_or_group.checkouts.sort_by { |x| x.sort_key(:due_date) }
```

I could add the safe operator here and I think we'd be all good?
```
    @checkouts = patron_or_group.checkouts&.sort_by { |x| x.sort_key(:due_date) }
```

Reply:
Thankfully only Faculty members are eligible - by longstanding policy - to have proxies.  I'm not saying it's never going to happen that a proxy is also a sponsor...but it shouldn't happen.